tests/check-style.py

```diff
@@ -0,0 +1,103 @@
+#! /usr/bin/env python3
+import os
+import ast
+import sys
+
+
+def check_number_literals_in_ast_Constant(node: ast.Constant, file_path: str) -> int:
+    """Check if large numeric literals use underscores as separators.
+
+    Args:
+        node: The AST node to check.
+        file_path: The path to the file containing the AST node.
+
+    Returns:
+        1 if the node is a large numeric literal without underscores, 0 otherwise.
+    """
+    if not isinstance(node.value, (int, float)):
+        return 0
+
+    # Get the original source code for the number
+    start_line, start_col = node.lineno, node.col_offset
+    end_line, end_col = node.end_lineno, node.end_col_offset
+    with open(file_path, "r") as f:
+        lines = f.readlines()
+    original_number = "".join(
+        line[start_col:end_col].strip() for line in lines[start_line - 1 : end_line]
+    )
+
+    sanitized_number = original_number
+    if "." in original_number:
+        # Only look at the part before the decimal point
+        sanitized_number = original_number[: original_number.index(".")]
+
+    if sanitized_number.startswith("0x"):
+        sanitized_number = sanitized_number[2:]
+
+    if "e" in sanitized_number.lower():
+        # Skip numbers with scientific notation
+        return 0
```

Comment:
I would also apply this to scientific notation: `1000e2` -> `1_000e2` (it could also cover for`1e1000` -> `1e1_000` but that's `inf`, although we might want to check it too if the code is the same or less complex).